src/main/java/it/pagopa/transactions/utils/AuthRequestDataUtils.java

```diff
@@ -0,0 +1,39 @@
+package it.pagopa.transactions.utils;
+
+import it.pagopa.ecommerce.commons.annotations.ValueObject;
+import it.pagopa.generated.transactions.server.model.OutcomeVposGatewayDto;
+import it.pagopa.generated.transactions.server.model.OutcomeXpayGatewayDto;
+import it.pagopa.generated.transactions.server.model.UpdateAuthorizationRequestDto;
+import it.pagopa.transactions.exceptions.InvalidRequestException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class AuthRequestDataUtils {
+
+    @ValueObject
```

Comment:
why @valueObject? is it a value object according to DDD? https://domaincentric.net/blog/ddd-building-blocks-in-php-value-object#:~:text=Value%20Object%20is%20an%20object,model%20behaviour%20of%20your%20application.

Reply:
in addition, does it make sense to have a util class for this? within the transactions domain? so in package `it/pagopa/transactions/utils` ?

Reply:
hopefully we will have only one gateway in the future so it sounds good to have this class so when we have only one gateway let's delete this class.